lib/List_solutions/row_with_maximum_ones_2643.ex

```diff
@@ -0,0 +1,29 @@
+defmodule RowAndMaximumOnes do
+  defstruct index: 0, previous_ones_count: 0
+  @spec row_and_maximum_ones(mat :: [[integer]]) :: [integer]
+  def row_and_maximum_ones(mat) do
+    mat
+    |> Enum.with_index()
+    |> max_ones(%RowAndMaximumOnes{})
+  end
+
+  defp max_ones([], %{index: index_num, previous_ones_count: prev_count} = info),
+    do: [index_num, prev_count]
+
+  defp max_ones(
+         [{list, index} | tail],
+         %{index: index_num, previous_ones_count: prev_count} = info
+       ) do
+    current_ones =
+      list
+      |> Enum.count(fn num -> num == 1 end)
+
+    info =
+      cond do
+        current_ones > prev_count -> %{info | index: index, previous_ones_count: current_ones}
+        true -> info
```

Comment:
According to the Elixir docs when you use COND you have to compulsory use the true keyword or else it gives an error about that and in CASE statements this is used